src/dmd/toobj.d

```diff
@@ -819,6 +816,8 @@ void toObjFile(Dsymbol ds, bool multiobj)
                 if (recurse(pd, isCtor) > 1)
                     pd.error("can only apply to a single declaration");
             }
+
+            visit(cast(AttribDeclaration)pd);
```

Comment:
Why did you move the call to the base class?

Reply:
The flag has to be set before visiting the function generation, because the pointers in init/fini are now written when visiting the function, which might be delayed with multiobj.
The problem was that they ended up in a different object file.